src/lib/user/user-actor.ts

```diff
@@ -44,4 +47,42 @@ export class UserActor {
     const { userMapper } = this.config;
     return userMapper.mapUserRecordToUserPii(record);
   }
+
+  public async getDogList(): Promise<DogList> {
+    const { dbPool, dogMapper } = this.config;
+    const sql = `
+    SELECT
+      dog_id as "dogId",
+      dog_encrypted_oii as "dogEncryptedOii",
+      CASE
+        WHEN dog_ever_pregnant = 'YES' THEN $5
+        WHEN dog_ever_received_transfusion = 'YES' THEN $5
+        WHEN dog_breed = '' AND dog_weight_kg IS NULL THEN $2
+        WHEN dog_weight_kg < 20 THEN $4
+        ELSE $3
+      END as "dogStatus"
+    FROM dogs
```

Comment:
Note: currently experimenting with views to associate each dog row with `dog_status`.

The view will also resolve the most recent `dog_weight_kg` value from the dog's record and its related reports. (This is needed to correctly evaluate the status.)